src/query/models/tags_test.go

```diff
@@ -226,18 +226,56 @@ func TestUpdateName(t *testing.T) {
 	actual, found := tags.Get(name)
 	assert.False(t, found)
 	assert.Nil(t, actual)
+	actual, found = tags.Name()
+	assert.False(t, found)
+	assert.Nil(t, actual)
 
 	value := []byte("n")
 	tags = tags.SetName(value)
 	actual, found = tags.Get(name)
 	assert.True(t, found)
 	assert.Equal(t, value, actual)
+	actual, found = tags.Name()
+	assert.True(t, found)
+	assert.Equal(t, value, actual)
 
 	value2 := []byte("abc")
 	tags = tags.SetName(value2)
 	actual, found = tags.Get(name)
 	assert.True(t, found)
 	assert.Equal(t, value2, actual)
+	actual, found = tags.Name()
+	assert.True(t, found)
+	assert.Equal(t, value2, actual)
+}
+
+func TestUpdateBucket(t *testing.T) {
+	name := []byte("!")
+	tags := NewTags(1, NewTagOptions().SetBucketName(name))
+	actual, found := tags.Get(name)
+	assert.False(t, found)
+	assert.Nil(t, actual)
+	actual, found = tags.Bucket()
+	assert.False(t, found)
+	assert.Nil(t, actual)
+
+	value := []byte("n")
+	tags = tags.SetBucket(value)
+	actual, found = tags.Get(name)
+	assert.True(t, found)
+	assert.Equal(t, value, actual)
+	actual, found = tags.Bucket()
+	assert.True(t, found)
+	assert.Equal(t, value, actual)
+
+	value2 := []byte("abc")
```

Comment:
Add test for empty val? e.g. `[]byte("")`

Reply:
Sure; not sure if it's necessary a special case though